### ISSUE TITLE
Feature Request: validate interger constant 

### ISSUE THREAD

**Issue by buckle2000** [rust-lang/rust#48351]
The following code should never compile.

```rust
fn main() {
    let sth: i32 = 100000000000; // or 100000000000i32
    println!("{}", sth); // 1215752192
}
```

Why rust compiler allows this?

**Comment by jonas-schievink** [comment]
It does warn about this:

```
warning: literal out of range for i32
 --> src/main.rs:2:20
  |
2 |     let sth: i32 = 100000000000; // or 100000000000i32
  |                    ^^^^^^^^^^^^
  |
  = note: #[warn(overflowing_literals)] on by default
```

And you can always `#[deny]` or `#[forbid]` the lint.

**Comment by buckle2000** [comment]
Well, it turns out that I was using an older version of rust.
